src/server/game/Entities/Player/Player.cpp

```diff
@@ -6391,11 +6409,17 @@ bool Player::UpdateFishingSkill()
 
     uint32 SkillValue = GetPureSkillValue(SKILL_FISHING);
 
-    int32 chance = SkillValue < 75 ? 100 : 2500 / (SkillValue - 50);
+    if (SkillValue >= GetMaxSkillValue(SKILL_FISHING))
+        return false;
 
-    uint32 gathering_skill_gain = sWorld->getIntConfig(CONFIG_SKILL_GAIN_GATHERING);
+    float chanceOfLevelUp = getProbabilityOfLevelUp(SkillValue);
 
-    return UpdateSkillPro(SKILL_FISHING, chance * 10, gathering_skill_gain);
+    /* Whenever the player clicks on the fishing gameobject the
+     * core will decide based on how many steps the player is
+     * into leveling up fishing, if the skill raises or not.
+     */
+    uint32 gathering_skill_gain = sWorld->getIntConfig(CONFIG_SKILL_GAIN_GATHERING);
```

Comment:
do we really need to create `gathering_skill_gain` ? as it's used only once, we could just use `sWorld->getIntConfig(CONFIG_SKILL_GAIN_GATHERING)` down there.

Reply:
... alternatively we can use `inline`